examples/next-prisma-starter-websockets/src/pages/_app.tsx

```diff
@@ -72,16 +73,17 @@ export default withTRPC<AppRouter>({
        * @link https://react-query.tanstack.com/reference/QueryClient
        */
       queryClientConfig: { defaultOptions: { queries: { staleTime: 60 } } },
-      headers: () => {
-        if (ctx?.req) {
-          // on ssr, forward client's headers to the server
-          return {
-            ...ctx.req.headers,
-            'x-ssr': '1',
-          };
-        }
-        return {};
-      },
+      headers: () =>
+        stripHeaders(() => {
+          if (ctx?.req) {
+            // on ssr, forward client's headers to the server
+            return {
+              ...ctx.req.headers,
+              'x-ssr': '1',
+            };
+          }
+          return {};
+        }),
```

Comment:
Can we skip the callback here? Makes it quite hard to read

```suggestion
      headers: () => {
        if (ctx?.req) {
          // on ssr, forward client's headers to the server
          return {
            ...stripHeaders(ctx.req.headers),
            'x-ssr': '1',
          };
        }
        return {};
      },
```

Reply:
@KATT More aesthetically pleasing, but less safe. This introduces the potential for runtime bugs when the developer forgets to wrap all returns with `stripHeaders` (in the same line of thinking: what if `x-ssr` is a reserved header? This would cause a server crash here), versus just having to wrap the entire `headers` function once.

Reply:
They can wrap the whole `return stripHeaders({})` as well, but if you _actually_ explicitly use one of the blocked headers it's better to have the server crash than to automagically strip it and lead people down a several hours down rabbit hole